pkg/cmd/server/api/v1/conversions.go

```diff
@@ -93,6 +98,56 @@ func init() {
 			out.KeyFile = in.ServerCert.KeyFile
 			return nil
 		},
+		func(in *NodeConfig, out *newer.NodeConfig, s conversion.Scope) error {
```

Comment:
shouldn't this do a default convert, then tweak the bits that don't happen automatically?